src/Lecture4.hs

```diff
@@ -200,7 +246,15 @@ row in the file.
 -}
 
 rowToStats :: Row -> Stats
-rowToStats = error "TODO"
+rowToStats (Row rProduct Buy rCost) = Stats
+  (Sum 1) (Sum $ -1 * rCost) (Max rCost) (Min rCost)
+  Nothing Nothing (Just $ Max rCost) (Just $ Min rCost)
+  (MaxLen rProduct)
+rowToStats (Row rProduct Sell rCost) = Stats
+  (Sum 1) (Sum rCost) (Max rCost) (Min rCost)
+  (Just $ Max rCost) (Just $ Min rCost) Nothing Nothing
+  (MaxLen rProduct)
+-- can this be written in a more readable & maintainable way?
```

Comment:
You can use record names to write this in a more maintainable way. So it'll be more clear what field goes where. You can also try to define a default record only once and then set only the different field to it.